blog.qmd

```diff
@@ -0,0 +1,228 @@
+
+```{r}
+corpus <- corpus(courses)
+```
+
+```{r}
+#| echo: false
+corpus[1:5]
+```
+
+```{r}
+tk <- tokens(corpus, what = "word", remove_punct = TRUE, remove_numbers = TRUE)
+```
+
+```{r}
+#| echo: false
+tk[1:5]
+```
+
+## Key-Term Searches with a Dictionary
+
+Our primary interest lies in identifying courses related to different types of work-based learning such as internships, apprenticeships, or practicums. For each type of work-based learning experience, we create a list of terms that we want to treat equivalently. For instance, our dictionary can specify that a course description refers to a clinical WBL if either of the terms "clinicals" or "clinical experience" appear.
+
+```{r}
+dict <-
+  dictionary(list(apprenticeship = "apprentice*",
+                  practicum = c("practicum", "practica"),
+                  coop = c("co-op", "cooperative education", "co-operative education"),
+                  clinicals = c('clinicals', 'clinical experience'),
+                  on_the_job = c("on the job training", "job training", "on-the-job training"),
+                  wbl = c('work-based learning', "work based learning", "wbl"),
+                  real_world_experience = c("real-world experience", "real world experience"),
+                  service_learning = "serive learning",
+                  field_experience = c("fieldwork", "field experience", "field-experience")))
+```
+
+Armed with our dictionary, we're ready to search for the terms using the kwic() (key-word in context) function. This function takes in a corpus and a dictionary as inputs, along with a window parameter specifying the number of tokens before and after a keyword that we want to see for context.
+
+```{r}
+keywords <- kwic(corpus, dict, window = 10) %>% as_tibble()
+```
+
+```{r}
+#| echo: false
+keywords %>% 
+  gt() %>% 
+  my_theme() %>% 
+  opt_interactive(use_text_wrapping = FALSE, use_resizers = TRUE, use_compact_mode = TRUE) 
+```
+
+Finally, we can join the results from the dictionary-based keyword in-context search back to the course level data and perform some wrangling to analyze the prevalence of work-based learning opportunities in Florida's community colleges.
+
+```{r}
+courses_with_keywords <-
+  keywords %>% 
+  left_join(courses, by = c("docname" = "doc_id")) %>%
+  mutate(sentence = glue("{pre} {keyword} {post}") |> str_trim() |> str_to_sentence()) %>% 
+  separate(docname, into = c("school", "course"), sep = "-", extra = 'merge'  ) %>% 
+  mutate(discipline = str_remove(discipline, ".* - ")) %>% 
+  select(school, discipline, course, statewide_course, degree_type, course_credits,  sentence, pattern) |>
+  arrange(school, pattern, discipline) %>% 
+  mutate(pattern = str_to_title(pattern) |> str_replace_all('_', '-'))
+```
+
+```{r}
+#| echo: false
+courses_with_keywords %>% 
+  gt() %>% 
+  my_theme() %>% 
+  opt_interactive(use_text_wrapping = FALSE, use_resizers = TRUE, use_compact_mode = TRUE) 
+```
+
+```{r}
```

Comment:
This is the main piece of output, so I'd have some text teeing it up. Maybe that means hiding the dataframe just above? But either way, I'd add a sentence explaining exactly how you generated this frequency table since this is the "climax" of the piece!

Reply:
@alenastern